### DISCUSSION TITLE
Inquiring about Bitwarden's Attitude Towards this Project

### DISCUSSION THREAD

**Discussion by gtf35** [dani-garcia/vaultwarden#3874] [Q&A] [answered]
Hello everyone! I have some questions regarding Bitwarden's attitude towards this project:

1. Does this project violate the license of Bitwarden?
2. What is Bitwarden's opinion and attitude towards this project?
3. Since this project implements some of the paid features of Bitwarden, I'm curious to know if it might lead to dissatisfaction from Bitwarden, or even causing them to collect and disclose the saved account passwords of users using the project?

I found some information on their [official community](https://community.bitwarden.com/t/bitwarden-vs-vaultwarden-rust-formerly-bitwarden-rs/15382), and it seems that they haven't explicitly expressed opposition. 

I'm not very good at legal knowledge and would like to hear your suggestions. Do I need to worry about this?

**Comment by kpfleming** [comment, marked as answer]
1. No.

3. Nobody here can speculate about the future behavior of the Bitwarden company or its staff. They do not have any way to 'collect and disclose' account passwords stored in Vaultwarden without implementing some sort of backdoor in their clients, which (when discovered) would pretty much be the end of their company as they would erase all trust they'd gathered prior to that point.

**Reply by gtf35** [reply]
Yeah, no one can predict their behavior.

As you said, their cost of collecting information about users using compatible servers is very high, it will consume users' trust in them.

But having said that, this does not really rule out this possibility, which is a bit regrettable (or I shouldn't say this, but...)

Taken together, I still choose to believe Bitwarden

**Comment by deajan** [comment]
I do think of Vaultwarden as a healthy alternative to Bitwarden for companies that would not spend money into such solutions anyways.
Bitwarden's target audience is still there, eg companies that want a support contract, guarantees etc.
Vaultwarden fills a nice gap for power users and smaller companies, which wouldn't pay Bitwarden anyways.

In the end Vaulwarden makes more indirect advertising for Bitwarden.
